### PR TITLE
test: add fuzz testing

### PR DESCRIPTION
**Pull request by zoruka**
https://nader.mirror.xyz/6Mn3HjrqKLhHzu2balLPv4SqE5a-oEESl4ycpRkWFsc

### CODE REVIEW

**Review by EmperorOrokuSaki — APPROVED** [review]
This looks very clean!

File: contracts/test/foundry/FleekERC721/GetToken.t.sol [hunk 1]
```diff
@@ -49,25 +58,23 @@ contract Test_FleekERC721_GetToken is Test_FleekERC721_Base {
...
 
-    function test_getTokenForDifferentAddresses() public {
-        vm.prank(address(1));
-        CuT.getToken(tokenId);
-        vm.prank(address(2));
-        CuT.getToken(tokenId);
-        vm.prank(address(3));
+    function testFuzz_getTokenForDifferentAddresses(address account) public {
+        vm.prank(account);
         CuT.getToken(tokenId);
```
Comment (jsonsivar) [inline contracts/test/foundry/FleekERC721/GetToken.t.sol:72]: Should we assert something here? This isn't related to new change but something I missed in a previous review.

File: contracts/test/foundry/FleekERC721/Mint.t.sol [hunk 2]
```diff
@@ -49,4 +49,22 @@ contract Test_FleekERC721_Mint is Test_FleekERC721_Base {
...
+        string memory appName,
+        string memory description,
+        string memory externalURL,
+        string memory ens,
+        string memory commitHash,
+        string memory gitRepository,
+        string memory logo,
+        uint24 color
+    ) public {
+        vm.assume(to != address(0));
```
Comment (jsonsivar) [inline contracts/test/foundry/FleekERC721/Mint.t.sol:64]: This makes me think, should we prevent mints that try to mint to addresss(0) on the contract?

File: contracts/test/foundry/FleekERC721/TokenURI.t.sol [hunk 3]
```diff
@@ -35,16 +35,14 @@ contract Test_FleekERC721_TokenURI is Test_FleekERC721_Base {
...
 
-    function test_tokenURIForDifferentAddresses() public {
-        vm.prank(address(1));
-        CuT.tokenURI(tokenId);
-        vm.prank(address(2));
-        CuT.tokenURI(tokenId);
-        vm.prank(address(3));
+    function testFuzz_tokenURIForDifferentAddresses(address account) public {
+        vm.prank(account);
         CuT.tokenURI(tokenId);
```
Comment (jsonsivar) [inline contracts/test/foundry/FleekERC721/TokenURI.t.sol:40]: Anything we can assert here?

**Review by jsonsivar — APPROVED** [review]
LGTM :fire:

**Pull request merged**